aten/src/ATen/native/sparse/SparseTensor.cpp

```diff
@@ -528,4 +528,63 @@ SparseTensor sparse_mask_cpu(const Tensor& t, const SparseTensor& mask) {
   return r;
 }
 
+Tensor sparse_mask_helper_cpu(
+  const SparseTensor& t,
+  const Tensor& mask_indices
+) {
+  /*
+    This is a helper function which filter values from `t._values()` using the `mask_indices`.
+    This CPU implementation uses a simple hash_map to filter values by matching the `mask_indices`
+    with the indices at tensor input `t`.
+
+    Inputs:
+      `t`             - coalesced sparse tensor input
+      `mask_indices`  - mask indices tensor
+
+    Note: every index exists in the sparse tensor `t`
```

Comment:
Is it a requirement? I don't think it's guaranteed

Reply:
Ohh my bad. You are right. it's not guaranteed